devito/parameters.py

```diff
@@ -45,3 +47,4 @@ def _updated(self, key, value):
 parameters = Parameters()
 # Default log level set to INFO
 parameters["log_level"] = 'INFO'
+parameters["autotuning"] = os.environ.get('DEVITO_ATMODE', 'basic')
```

Comment:
I think you're right, having functions is better

But how do we decide whether to populate from "config file" rather than "env_vars+defaults"? Do we use a special env var ?